File: frontend/src/types/photo.ts
// ============================================================================
// Section A: Generated type aliases from OpenAPI schema
// Source of truth: backend Pydantic models -> openapi.json -> api.generated.ts
// Run `bun run generate:api` after backend schema changes and commit both files.
// ============================================================================

import type { components } from './api.generated'

export type PhotoUpdate = components['schemas']['PhotoUpdate']

// ============================================================================
// Section B: Hand-maintained frontend-only types
// Photo routes have no response_model; they return custom dict payloads
// from PhotoService.build_photo_payload(). These match the frontend shape,
// not a generated schema.
// ============================================================================

export interface Photo {
  id: number
  filename: string
  path: string
  thumbnail_url?: string | null
  size: number
  is_main: boolean
  caption?: string | null
  uploaded_at?: string
}

export interface PhotoListResponse {
  photos: Photo[]
  total: number
}

File: backend/app/schemas/photo.py
"""
Photo schemas for Pydantic validation
"""

from datetime import datetime as datetime_type

from pydantic import BaseModel, Field


class PhotoResponse(BaseModel):
    """Response model for photo"""

    id: int
    vin: str
    file_path: str
    thumbnail_path: str | None = None
    thumbnail_url: str | None = None
    is_main: bool
    caption: str | None = None
    uploaded_at: datetime_type

    class Config:
        from_attributes = True


class PhotoListResponse(BaseModel):
    """Response model for photo list"""

    photos: list[PhotoResponse]
    total: int
    main_photo: PhotoResponse | None = None


class PhotoUpdate(BaseModel):
    """Schema for updating photo"""

    caption: str | None = Field(None, max_length=200)
    is_main: bool | None = None
